sdk-base/src/main/java/com/mapbox/maps/util/MapboxExceptions.kt

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.maps.util
+
+/**
+ * Mapbox exception class flagging map related [RuntimeException] did occur.
+ */
+class MapboxMapException(
```

Comment:
Have you considered commonly used `IllegalArgumentException` / `IllegalStateException` instead of custom ones? Does custom `MapboxMapException` actually gives more information about the exception?

Reply:
Changing exception type is a breaking change, we can't afford doing it.

Custom annotation type (inherited from `RuntimeException`) may be useful if for some reason runtime exception is thrown from our SDK but user is OK to simply catch it (e.g. adding a layer that was already added). This way user could listen precisely to Mapbox style exception and not catch general `RuntimeException`.

Additional pro is actually having a clearer stacktrace showing what part of our SDK did throw an exception.

Reply:
> Changing exception type is a breaking change, we can't afford doing it.

Both `IllegalArgumentException` / `IllegalStateException` are inherited from `RuntimeException`, what's the difference from our own inherited exception then?

Reply:
We could use `sealed class` implementation here, let `MapboxMapException` be a sealed class and other below classes inherit from that.

Reply:
@ank27 sealed class is useful when user will be using `when` clause which is not the case here. However sealed classes are not greatly supported in Java, so let's stick with current approach.

Reply:
@yunikkk as discussed offline let's stick to following schema: custom Mapbox exception is thrown mostly in cases when native method returns `Error` and not `Value` while if our Android SDK code is throwing some exception - we will stick to `IllegalArgumentException` / `IllegalStateException` depending on the situation.
cc @mapbox/maps-android